src/tests/ftest/control/dmg_telemetry_io_basic.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from avocado.core.exceptions import TestFail
+
+from ior_test_base import IorTestBase
+from telemetry_test_base import TestWithTelemetry
+
+
+class TestWithTelemetryIOBasic(IorTestBase,TestWithTelemetry):
+    # pylint: disable=too-many-ancestors
+    # pylint: disable=too-many-nested-blocks
+    """Test telemetry engine io basic metrics.
+
+    :avocado: recursive
+    """
+
+
+    def verify_io_test_metrics(self, io_test_metrics, metrics_data, threshold):
+        """ Verify telemetry io metrics from metrics_data.
+
+        Args:
+            io_test_metrics (list): list of telemetry io metrics.
+            metrics_data (dict): a dictionary of host keys linked to a
+                                 list of io metric names.
+            threshold (int): test io metrics threshold.
+
+        """
+        status = True
+        for name in sorted(io_test_metrics):
+            self.log.info("  --telemetry metric: %s", name)
+            self.log.info(
+                "    %-9s %-12s %-4s %-6s %-6s %s",
+                "TestLoop", "Host", "Rank", "Target", "Size", "Value")
+            for key in sorted(metrics_data):
+                m_data = metrics_data[key]
+                if key == 0:
+                    testloop = "Initial"
+                else:
+                    testloop = str(key)
+                for host in sorted(m_data[name]):
+                    for rank in sorted(m_data[name][host]):
+                        for target in sorted(m_data[name][host][rank]):
+                            for size in sorted(
+                                m_data[name][host][rank][target]):
+                                value = m_data[name][host][rank][target][size]
+                                invalid = ""
+                                #Verify value within threshold
+                                if (value < threshold[0]
+                                    or value >= threshold[1]):
+                                    status = False
+                                    invalid = "*exceed threshold"
```

Comment:
If I understand correct, we are checking for the `value` to be within the range of `threshold[0]` and `threshold[1]`. In that case `invalid = "*exceed threshold"` can be a bit misleading. So something like `invalid = "*out of range"` will make more sense.

Reply:
The telemetry io threshold is changing depended on the metrics and io transactions, this will be continue to be verified. I will keep note if this is the correct definition.
At this phase, we raise failure in case the value is outside the threshold limit.